### PR TITLE
[TESTING] Adjust head behaviour within soccer strategy to better deal with seeing ball and goals simultaneously

### PR DESCRIPTION
**Pull request by MattAmos**

### CODE REVIEW

File: module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp [hunk 1]
```diff
@@ -316,7 +320,7 @@ namespace behaviour {
             }
             else*/ if (NUClear::clock::now() - ballLastMeasured
```
Comment (JakeFountain) [inline module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp]: delete else comment

File: module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp [hunk 2]
```diff
@@ -488,6 +492,24 @@ namespace behaviour {
                     default: throw std::runtime_error("Soccer strategy attempted to find a bad object");
                 }
             }
+
+            // When weve seen both field targets
+            if (soccerObjectPriority->ball && soccerObjectPriority->goal) {
+                // Check field message and localisation ball message
+                // Check trace of their covariances are bigger than configurable threshold
```
Comment (JakeFountain) [inline module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp]: check for ball proximity here

File: module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp [hunk 3]
```diff
@@ -325,12 +329,12 @@ namespace behaviour {
                         && (Eigen::Vector2d(field.position[0], field.position[1]).norm()
                             > 1)) {  // a long way away from centre
                         // walk to centre of field
-                        find({FieldTarget(FieldTarget::Target::BALL)});
```
Comment (JakeFountain) [inline module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp]: use ball and goal search more often

**Comment by ysims** [comment]
Closing PR, old/outdated but could be worth rewriting in the future.

**Pull request closed without merging**